packages/cli/uni-builder/src/shared/parseCommonConfig.ts

```diff
@@ -263,6 +272,30 @@ export async function parseCommonConfig(
     });
   }
 
+  if (tagsByEntries) {
+    extraConfig.html.tags = [
+      (tags, utils) => {
+        const entryTags = castArray(tagsByEntries[utils.entryName]);
+
+        const handlers: HtmlTagHandler[] = [];
+
+        for (const tag of entryTags) {
+          if (isFunction(tag)) {
+            // The function will be executed at the end of the HTML processing flow
+            handlers.push(tag);
+          } else {
+            tags.push(tag);
+          }
+        }
+
+        return handlers.reduce(
+          (currentTags, handler) => handler(currentTags, utils) || currentTags,
```

Comment:
it is unexpected to run the handlers inside the uni-builder.

uni-builder just need to:

- transform `html.tagsByEntries` to `html.tags`
- sort the `html.tags` array, put functions at last

Reply:
we can't sort and put functions directly, because we should filter `entries` by `utils.entryName` first, which can only be geted in `HtmlInjectTagHandler`, and `HtmlInjectTagHandler` does not support returning functions/arrays.